File: data-structure-and-algorithm/jian-zhi-offer/0005.Replace-Space/ReplaceSpace.java
public class Main05 {
	static String replaceSpace(StringBuffer str) {
		for (int i = 0; i < str.length(); i++) {
			if (str.charAt(i) == ' ') {
				str.replace(i, i + 1, "%20");
			}
		}
		return new String(str);
	}

	public static void main(String[] args) {
		System.out.println(replaceSpace(new StringBuffer("Hello World")));
	}
}

File: data-structure-and-algorithm/jian-zhi-offer/0005.Replace-Space/ReplaceSpace.py

class QueueByTwoStacks:
    def replacespace(self, s: str) -> str:
        result = []
        for i in s:
            if i == ' ':
                result.append("%20")
            else:
                result.append(i)
        return "".join(result)


if __name__ == '__main__':
    case1 = "Hello world"

    print(Main05().replacespace(case1))
